src/cmd/api/router/router.go

```diff
@@ -39,3 +49,38 @@ func getCORS() (handlers.CORSOption, handlers.CORSOption, handlers.CORSOption, h
 
 	return corsObj, methods, headers, credentials
 }
+
+// Metrics
```

Comment:
I'm not sure to understand what is done with the middleware and the NewResponseWrite, can you add a bit of context/ some comments ?